externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala

```diff
@@ -59,6 +63,17 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
         spark.newSession()
       }
 
+      if (singleSparkSessionInitialized.compareAndSet(false, true)
```

Comment:
if singleSparkSessionInitialized then we need always updating?

Reply:
no, `compareAndSet(expect = false, update = true)` only return `true` in first round, but the if statement break the semantic of `singleSparkSessionInitialized ` here, it should be
```
if (!this.conf.get(ENGINE_SINGLE_SPARK_SESSION)) || singleSparkSessionInitialized.compareAndSet(false, true))
```
anyway, removed `singleSparkSessionInitialized` and partial move the init logic to engine startup